File: java/src/main/java/fizz/buzz/Action.java
package fizz.buzz;

public interface Action {
    String to(int n);
    
    static Action to(String str) {
        return new Action() {
            @Override
            public String to(int n) {
                return str;
            }            
        };
    }
        
    static Action nop() {
        return new Action() {
            @Override
            public String to(int n) {
                return Integer.toString(n);
            }
        };
    }
}

File: cpp/src/fizz-buzz-whizz/Action.cpp
#include "fizz-buzz-whizz/Action.h"

Action to(std::string&& str) {
  return [str = std::move(str)](auto) {
    return str;
  };
}

Action nop() {
  return [](auto n) {
    return std::to_string(n);
  };
}
